### PR TITLE
[Scheduler User Guide Documentation] Add user guide documentation abo…

### PR DESCRIPTION
**Pull request by wgliang**
…ut how to customize and extend the kube-scheduler

The default scheduler satisfies most of our scheduling requirements, but sometimes we have to modify the default scheduler to meet our scheduling needs, depending on the production environment and the scenario. So at this time we need to customize and extend kube-scheduler.

The PR if from https://github.com/kubernetes/kubernetes/issues/69769

/sig scheduling
/kind documentation

### CODE REVIEW

**Comment by Rajakavitha1** [comment]
@bsalamat could you please review the updated content.

File: content/en/docs/concepts/scheduling/customization-and-extension.md [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+  - ISCSI forbids if any two pods share at least same IQN, LUN and Target
+- `GeneralPred`: Checks whether noncriticalPredicates and EssentialPredicates pass. noncriticalPredicates are the predicates that only non-critical pods need and EssentialPredicates are the predicates that all pods, including critical pods, need.
+- `CheckNodeMemoryPressure`: Checks if a pod can be scheduled on a node reporting memory pressure condition.
+- `CheckNodePIDPressure`: Checks if a pod can be scheduled on a node reporting pid pressure condition.
+- `CheckNodeDiskPressure`: Checks if a pod can be scheduled on a node reporting disk pressure condition.
+- `CheckNodeCondition`: Checks if a pod can be scheduled on a node reporting out of disk, network unavailable and not ready condition. Only node conditions are accounted in this predicate.
+- `PodToleratesNodeTaints`: checks if a pod tolerations can tolerate the node taints.
+- `CheckVolumeBinding`: Evaluates if a pod can fit due to the volumes it requests, for both bound and unbound PVCs.
+- CheckNodeUnschedulablePred: Checks if a pod can be scheduled on a node with Unschedulable spec.
```
Comment (bsalamat) [inline content/en/docs/concepts/scheduling/customization-and-extension.md]: Please use the same formatting for the predicate names, to be consistent with the above.

File: content/en/docs/concepts/scheduling/customization-and-extension.md [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+- CheckServiceAffinity: It is a predicate which matches nodes in such a way to force that ServiceAffinity.labels are homogenous for pods that are scheduled to a node. (i.e. it returns true IFF this pod can be added to this node such that all other pods in the same service are running on nodes with the exact same ServiceAffinity.label values).(WARNING: This Predicate is NOT guaranteed to work if some of the predicateMetadata data isn't precomputed... For that reason it is not exported, i.e. it is highly coupled to the implementation of the FitPredicate construction.)
+- MatchInterPodAffinity: Checks if a pod can be scheduled on the specified node with pod affinity/anti-affinity configuration. First return value indicates whether a pod can be scheduled on the specified node while the second return value indicates the predicate failure reasons if the pod cannot be scheduled on the specified node.
+- PodFitsHost: Checks if a pod spec node name matches the current node.
+
+{{< note >}}
+**Note:** Bold display is the default predicates.
```
Comment (bsalamat) [inline content/en/docs/concepts/scheduling/customization-and-extension.md]: None of them are bold

**Comment by Rajakavitha1** [comment]
@wgliang could you please look into the suggestions that @bsalamat has made on this PR.

**Comment by zparnold** [comment]
@wgliang Closing this, please let me know when you'd like to continue! 😄 We'd love to have your contribution!

**Comment by wgliang** [comment]
@zparnold Sorry, this PR is being reviewing and updating. 
/reopen

**Comment by mistyhacks** [comment]
What's the status of this PR? It is now 28 days since the last comment, which was to reopen after the PR was previously closed due to no activity.

**Comment by Rajakavitha1** [comment]
What's the status of this PR? It is over a month since the last comment.
@wgliang  Please let us know if you would like to close this PR.

**Comment by wgliang** [comment]
@Rajakavitha1 Sorry, reviewing took too long, I will close it until we need to reopen it.
/close

**Comment by sftim** [comment]
This work might be useful in connection with #14637.

**Pull request closed without merging**